xml_callbacks.go

```diff
@@ -0,0 +1,56 @@
+package libxml2
+
+import (
+	"strings"
+
+	"github.com/lestrrat-go/libxml2/clib"
+)
+
+type XMLCallback struct {
+	uri         string
+	matchingFnc func(receivedURI, expectedURI string) bool
+	data        []byte
+}
+
+func (i XMLCallback) CanHandle(uri string) bool {
+	return i.matchingFnc(uri, i.uri)
+}
+
+func (i XMLCallback) GetData(_ string) []byte {
+	return i.data
+}
+
+var _ clib.XMLCallback = (*XMLCallback)(nil)
+
+type Option func(*XMLCallback)
```

Comment:
You should not be defining your own `type Option` when it's already using  the other "option" pattern.

Reply:
Reverted